### PR TITLE
(maint) Update powershell helpers to escape correctly for linux

### PR DESCRIPTION
**Pull request by McdonaldSeanp**
This commit updates the PuppetBolt powershell module to correctly escape hash
parameters for linux platforms. Conversation in this thread:
https://github.com/MicrosoftDocs/PowerShell-Docs/issues/2361 describe the
circumstances of why interpolation breaks.

The fix is to include backslash escape chars for all quotes in hash parameters
if the powershell function is running on linux

### CODE REVIEW

File: pwsh_module/pwsh_bolt_internal.ps1 [hunk 1]
```diff
@@ -42,7 +42,19 @@ function Get-BoltCommandline {
...
+      #
+      # See https://github.com/MicrosoftDocs/PowerShell-Docs/issues/2361
+      #
+      # $IsLinux is an automatic variable from powershellcore:
+      # https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-7.1#islinux
+      if ($IsLinux) {
+        $params += $v -replace '"', '\"'
+      } else {
+        $params += "'$($v)'"
+      }
```
Comment (jpogran) [inline pwsh_module/pwsh_bolt_internal.ps1:57]:
```suggestion
     $IsLinuxEnv = (Get-Variable -Name "IsLinux" -ErrorAction Ignore) -and $IsLinux
     $IsMacOSEnv = (Get-Variable -Name "IsMacOS" -ErrorAction Ignore) -and $IsMacOS
     $IsWinEnv = !$IsLinuxEnv -and !$IsMacOSEnv
     if($IsWinEnv){
       $params += "'$($v)'"
     } else {
       $params += $v -replace '"', '\"'
     }
```

Comment (jpogran) [reply pwsh_module/pwsh_bolt_internal.ps1:57]: I think we'll need the get-variable pattern to handle being run in 5.1, and we'll need to account for macox not being detected as linux.

**Review by jpogran — CHANGES_REQUESTED** [review]

File: pwsh_module/pwsh_bolt_internal.ps1 (hunk 1 above)
Comment (McdonaldSeanp) [reply pwsh_module/pwsh_bolt_internal.ps1]: Updated

**Review by jpogran — APPROVED** [review]

**Comment by jpogran** [comment]
@McdonaldSeanp just need to update CI to run these tests on nix and macos, and adjust the pester tests to account for the text changes and we're good to go!

**Pull request merged**